src/main/java/com/googleintern/wfm/ruleengine/model/UserPoolAssignmentModel.java

```diff
@@ -0,0 +1,37 @@
+package src.main.java.com.googleintern.wfm.ruleengine.model;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableSet;
+import java.util.Set;
+
+@AutoValue
+public abstract class UserPoolAssignmentModel {
+    public abstract int userId();
+
+    public abstract int workforceId();
+
+    public abstract int workgroupId();
+
+    public abstract ImmutableSet<FilterModel> filters();
```

Comment:
If this UserPoolAssignmentModel represent a user, then a filter is not an appropriate property of a user. 
A user could have a skill id or a role id associate with him/her but he/she cannot have a "filter".

Reply:
I am not sure about this comment. The Filter class supposes to store role ids, skills, role skill ids. I  changed the filterType to a enum variable(to identify role ids, skills, role skill ids). The value stores the integer id.

Reply:
I mean a user should not have a FilterModel as his property. Filters can be property of Rule but it can never be a property of a user. 
Instead you should have ImmutableList<Long> skillIds, ImmutableList<Long> roleIds ... as properties for a user.